server/apps/publish/archive_publish.py

```diff
@@ -282,48 +302,92 @@ def publish(self, doc, updates, target_output_channels=None):
                 updates['source'] = desk['source'] if desk and desk.get('source', '') \
                     else DEFAULT_SOURCE_VALUE_FOR_MANUAL_ARTICLES
 
+            updates['pubstatus'] = 'canceled' if self.publish_type == 'killed' else 'usable'
+            updated.update(updates)
+
+        # Step 2(a)
+        if self.published_state == 'killed':
+            queued_items = get_resource_service('publish_queue').get(
+                req=None, lookup={'item_id': updated[config.ID_FIELD]})
+
+            if queued_items.count():
+                # Step 2(a)(i)
+                subscribers = self._find_subscribers(queued_items)
+                recipients = [s['email'] for s in subscribers]
+                send_article_killed_email(doc, recipients)
```

Comment:
I think the killed email should be sent to all subscribers, not only those who received the original.